### DISCUSSION TITLE
How to embed an external widget

### DISCUSSION THREAD

**Discussion by phoskee** [zauberzeug/nicegui#2602] [Q&A]
### Question

I am trying to embed a widget, consisting of a` <script>` and an `<a>`. 
Now, if I put the code in `ui.add_body_html()`, the widget works and is displayed. 
The only problem is that it is shown at the top of the page, on every page. 
I have tried to do this, with no valid results:
```py
ui.add_body_html(r'''<script src="https://cdn.logwork.com/widget/countdown.js"></script> ''')
ui.html('''<a class="countdown-timer" data-timezone="Europe/Rome" data-language="it" data-textcolor="#ffffff" data-date="2024-07-26 10:00" data-background="#ffffff" data-digitscolor="#000000" data-unitscolor="#ffffff">ROAD TO PARIS 2024</a>''').classes('countdown-timer')
```

**Comment by falkoschindler** [comment]
Hi @phoskee,

Can you help us to understand how countdown.js would normally be used? I can't immediately find documentation for this library. How would a plain HTML file look like?

**Reply by phoskee** [reply]
I honestly don't know how it works, I just know that through [this site](https://logwork.com/countdown-timer) you can create the widget and then embed in the html

**Comment by falkoschindler** [comment]
Ok, I don't know what I did differently. But when copying the code from https://logwork.com/countdown-timer "Embed on your website" and wrap it like this
```py
ui.add_body_html('<script src="https://cdn.logwork.com/widget/countdown.js"></script>')
ui.html('<a href="https://logwork.com/countdown-timer" class="countdown-timer" data-timezone="Europe/Berlin" data-date="2024-02-28 16:18">Countdown Timer</a>')
```
it works nicely:
<img width="254" alt="Screenshot 2024-02-26 at 16 22 28" src="https://github.com/zauberzeug/nicegui/assets/5767091/040abea3-1b46-4e67-9a1b-b763c7c71fde">

**Reply by phoskee** [reply]
Found the issue! I put the code to a running app, but if I save and then run, everything works. 
Because of this "issue" I had to disable the tab refresh. 
With each refresh everything breaks and I have to restart the app, but that's okay!
